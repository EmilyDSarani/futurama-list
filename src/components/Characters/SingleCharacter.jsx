//this will literally just render how the detail page is going to look
import React from 'react'
import styles from './Character.css'

export default function SingleCharacter({ character }) {
    const{Name, PicUrl, Species, Status } = character

    console.log(character)
    return (
        <div className={styles.character}>
            <img src={PicUrl} alt={Name} />
            <h1 className={styles.content}>Name: {Name}</h1>
            <h2>Species: {Species}</h2>
            <h3>Status: {Status}</h3>
        <div className={styles.bg}></div>
        <div className={styles.bg2}></div>
        <div className={styles.bg3}></div>
        
        </div>
        
    )
}
