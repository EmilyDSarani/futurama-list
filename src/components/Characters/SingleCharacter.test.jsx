import { render } from '@testing-library/react'
import SingleCharacter from './SingleCharacter'

it('should render character', () => {
    const { container } = render(
    <SingleCharacter character={{
        image: 'https://place-puppy.com/',
        name:'Sarani',
        species:'Humanoid',
        status:'possibly alive'}} />);
    expect(container).toMatchSnapshot();
});