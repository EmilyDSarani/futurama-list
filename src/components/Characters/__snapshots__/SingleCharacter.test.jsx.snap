// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render character 1`] = `
<div>
  <div
    class="character"
  >
    <h1
      class="headerName"
    >
      Character:
    </h1>
    <img />
    <h1
      class="content"
    >
      Name: 
    </h1>
    <h2>
      Species: 
    </h2>
    <h3>
      Status: 
    </h3>
    <div
      class="bg"
    />
    <div
      class="bg2"
    />
    <div
      class="bg3"
    />
  </div>
</div>
`;
